algorithms/kernel/k_nearest_neighbors/oneapi/bf_knn_classification_predict_kernel_ucapi_impl.i

```diff
@@ -146,20 +123,15 @@ Status KNNClassificationPredictKernelUCAPI<algorithmFpType>::
 
     const size_t nDataBlocks = nDataRows / dataBlockSize + int (nDataRows % dataBlockSize != 0);
     const size_t nProbeBlocks = nProbeRows / probeBlockSize + int (nProbeRows % probeBlockSize != 0);
-    QuickSelectIndexed::Result selectResult(context, nK, probeBlockSize, distances.type(), categories.type(), &st);
+    SelectIndexed::Result selectResult(context, nK, probeBlockSize, distances.type(), &st);
```

Comment:
I do not understand why you need to create `selectResult` right here and why do you need to create it explicitly rather than to return from some function.